packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -63,7 +63,11 @@ export default class ConfigParser {
          * rather than relative to the config file
          */
         if (_initialConfig.spec) {
-            _initialConfig.spec = makeRelativeToCWD(_initialConfig.spec) as string[]
+            // Find dir path for single spec using glob pattern
+            const specsDirPath = _initialConfig.spec[0].includes(path.sep) || _initialConfig.spec[0].includes('/')
+                ? process.cwd()
+                : path.dirname(path.join(process.cwd(), _pathService.glob(_initialConfig.spec[0], process.cwd())[0]))
+            _initialConfig.spec = makeRelativeToCWD(_initialConfig.spec, specsDirPath) as string[]
```

Comment:
I don't really understand this, e.g. the following section:

```
_initialConfig.spec[0].includes(path.sep) || _initialConfig.spec[0].includes('/')
```

Why do we need to check for both here? In which cases is `path.sep` not sufficient?

Also we are only looking at `_initialConfig.spec[0]` but what if there are multiple specs in this array from different directories?

Reply:
Agreed, and correct me if I am wrong but I am pretty sure the specs should be relative to the config and not use process.cwd()